### PR TITLE
Scrub XML response stream of any prepended junk

### PR DESCRIPTION
**Pull request by tonyr59h**
Also adds Robolectric unit testing support to library project.

Not ready for review until I move tests to the example project.

cc @maxme

### CODE REVIEW

**Comment by maxme** [comment]
@tonyr59h Should we move the test to the example project?

**Comment by tonyr59h** [comment]
> Should we move the test to the example project?

Yup 👍  Moved in b441d583c2918d7f2c018943143f50440971e2ad

File: example/src/test/java/org/wordpress/android/fluxc/utils/XMLSerializerUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            byte[] bb = new byte[xmlLength];
+            int val;
+            for (int i = 0; i < bb.length && ((val = resultStream.read()) != -1); ++i) {
+                bb[i] = (byte) val;
+            }
+
+            is.close();
+            resultStream.close();
+
+            return new String(bb, "UTF-8");
```
Comment (maxme) [inline example/src/test/java/org/wordpress/android/fluxc/utils/XMLSerializerUtilsTest.java]: Java... 😞

**Comment by maxme** [comment]
I've added a "real life" test in 39a79aa
:shipit:

**Pull request merged**